Getting no-unused-vars errors on function arguments, even with args="none"

Version 0.6.2

Even with no-unused-vars set to args=none, eslint is still throwing errors for some unused arguments in both server-side and client-side code.

It seems to only occur when _no_ arguments are used. When the function has two args, and neither are used, they both give `no-unused-var` errors. But once one of them is used, regardless of which one, the errors disappear.
# 

```
// eslint.rc
{
  "no-unused-vars": [2, {"args": "none"}],
  // ...
}
```

```
// public/js/services/cookies.js
Application.addService('cookies', function(application) {
  // application is un-used here
  // ... 
});
```

```
// test output
public/js/services/cookies.js
  13:47  error  application is defined but never used  no-unused-vars
public/js/services/templates.js
  13:49  error  application is defined but never used  no-unused-vars
public/js/modules/menu-list.js
  15:48  error  context is defined but never used  no-unused-vars
```

--- a/lib/rules/no-unused-vars.js
+++ b/lib/rules/no-unused-vars.js
@@ -48,6 +48,24 @@ module.exports = function(context) {
         return candidates;
     }
 
+    /**
+     * Check if funcNode has paramNode as an identifier of one of its params.
+     *
+     * @param {FunctionExpression|FunctionDeclaration} funcNode Node which to
+     *                                                          check for param
+     * @param {Identifier} paramNode Node which can be an identifier for a param
+     * @returns {boolean} If funcNode has paramNode as an identifier of one of
+     *                    its params
+     */
+    function functionHasParam(funcNode, paramNode) {
+        for (var i = 0, len = funcNode.params.length; i < len; i++) {
+            if (funcNode.params[i] === paramNode) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     function populateVariables(node) {
         var scope = context.getScope(),
             functionName = node && node.id && node.id.name;
@@ -66,7 +84,8 @@ module.exports = function(context) {
                     variables[variable.name].push({
                         name: variable.name,
                         node: variable.identifiers[0] || node,
-                        used: (variable.name === functionName)
+                        used: (variable.name === functionName),
+                        ignorable: (config.args === "none") && node.params && functionHasParam(node, variable.identifiers[0])
                     });
                 }
             }
@@ -108,24 +127,6 @@ module.exports = function(context) {
      */
     function isFunction(node) {
         return node && node.type && (node.type === "FunctionDeclaration" || node.type === "FunctionExpression");
-    }
-
-    /**
-     * Check if funcNode has paramNode as an identifier of one of its params.
-     *
-     * @param {FunctionExpression|FunctionDeclaration} funcNode Node which to
-     *                                                          check for param
-     * @param {Identifier} paramNode Node which can be an identifier for a param
-     * @returns {boolean} If funcNode has paramNode as an identifier of one of
-     *                    its params
-     */
-    function functionHasParam(funcNode, paramNode) {
-        for (var i = 0, len = funcNode.params.length; i < len; i++) {
-            if (funcNode.params[i] === paramNode) {
-                return true;
-            }
-        }
-        return false;
     }
 
     function markIgnorableUnusedVariables(usedVariable) {

--- a/tests/lib/rules/no-unused-vars.js
+++ b/tests/lib/rules/no-unused-vars.js
@@ -41,6 +41,7 @@ eslintTester.addRuleTest("lib/rules/no-unused-vars", {
         { code: "function g(bar, baz) { return bar; }; g();", args: [1, {"vars": "all", "args": "none"}] },
         { code: "function g(bar, baz) { return 2; }; g();", args: [1, {"vars": "all", "args": "none"}] },
         { code: "function g(bar, baz) { return bar + baz; }; g();", args: [1, {"vars": "locals", "args": "all"}] },
+        { code: "var g = function (bar, baz) { return 2; }; g();", args: [1, {"vars": "all", "args": "none"}] },
         "(function z() { z(); })();",
         { code: " ", globals: {a: true} }
     ],

